benchmarks/common.py

```diff
@@ -1530,145 +1395,73 @@ def main(runner, original_dir=None):
-        backend_str = "prims_nvfuser"
-        output_filename = f"accuracy_aot_{backend_str}.csv"
-    elif args.print_fx:
-        optimize_ctx = torchdynamo.optimize(
-            print_fx,
-            nopython=args.nopython,
-        )
-    elif args.print_aten_ops:
-        optimize_ctx = torchdynamo.optimize(
-            print_aten_ops,
-            nopython=args.nopython,
-        )
-    elif args.accuracy_ts:
-        optimize_ctx = torchdynamo.optimize(fixed_strategy1, nopython=args.nopython)
-        experiment = speedup_experiment
-        backend_str = "nvfuser" if args.nvfuser else "nnc"
-        output_filename = f"accuracy_{backend_str}.csv"
-    elif args.nothing:
-        pass
-    elif args.nops:
-        optimize_ctx = torchdynamo.eval_frame._optimize_catch_errors(
-            torchdynamo.testing.debug_insert_nops, nopython=args.nopython
-        )
-    elif args.backend:
-        optimize_ctx = torchdynamo.optimize(args.backend, nopython=args.nopython)
-        experiment = speedup_experiment
-        output_filename = f"speedup_{args.backend}.csv"
-    elif args.log_conv_args:
-        optimize_ctx = torchdynamo.optimize(conv_args_analysis, nopython=args.nopython)
-        output_filename = "log_conv_args.csv"
-    elif args.recompile_profiler:
-        output_filename = "recompile_profiler_log.csv"
-        experiment = recompile_profiler_experiment
-    else:
-        optimize_ctx = torchdynamo.optimize(fx_insert_profiling, nopython=args.nopython)
-        experiment = coverage_experiment
-        output_filename = "coverage.csv"
+
+    # these names are based on the historical command line flag values
+    backends = {
+        "overhead": dummy_fx_compile,
+        "online_autotune": online_autotuner,
+        "offline_autotune": offline_autotuner,
+        "python_key": python_key,
+        "ltc": backends.ltc_reuse_graph,
+        "ltc_trivial": backends.ltc_trivial,
+        "fixed1": fixed_strategy1,
+        "fixed2": fixed_strategy2,
+        "ts": backends.ts,
+        "fx2trt": backends.fx2trt_compiler,
+        "fx2trt_fp16": backends.fx2trt_compiler_fp16,
+        "print_fx": print_fx,
+        "print_aten_ops": print_aten_ops,
+        "nops": torchdynamo.testing.debug_insert_nops,
+        "log_conv_args": conv_args_analysis,
+        "fx_insert_profiling": fx_insert_profiling,
+    }
+
+    experiments = {
+        "speedup": speedup_experiment,
+        "cold_start": cold_start_experiment,
+        "speedup_ts": speedup_experiment_ts,
+        "speedup_sr": speedup_experiment_sr,
+        "speedup_onnx": speedup_experiment_onnx,
+        "speedup_trt": speedup_experiment_trt,
+        "speedup_fx2trt": speedup_experiment_fx2trt,
+        "recompile_profiler": recompile_profiler_experiment,
+        "coverage": coverage_experiment,
+    }
+
+    frontends = {
+        "optimize": torchdynamo.optimize,
+        "optimize_catch_errors": torchdynamo.eval_frame._optimize_catch_errors,
+    }
```

Comment:
This might not capture configuration that use
- eager mode (`--nothing`): useful for debugging issues in torchbench, when it break up stream
- torchscript: baselines to compare against

Reply:
Do we really care about running against TS? I feel like the only reason we would try it is if we were trying to write a blog post comparing against TS, but this seems like a lot of work for not much publicity gain.

Reply:
Baselines are useful, and when we release more broadly I bet there will be an audience interested in comparing to TorchScript.